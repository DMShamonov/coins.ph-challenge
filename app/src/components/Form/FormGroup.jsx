// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { FormGroupPropsType } from './types.js.flow';

export default function FormGroup({
  className,
  children,
}: FormGroupPropsType): React.Node {
  return (
    <div className={classnames('form__group', className)}>{children}</div>
  );
}
