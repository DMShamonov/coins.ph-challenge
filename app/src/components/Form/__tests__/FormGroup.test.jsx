import React from 'react';
import { shallow } from 'enzyme';

import FormGroup from '../FormGroup';

describe('`FormGroup` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<FormGroup>__CONTENT__</FormGroup>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<FormGroup className="__CLASSNAME__">__CONTENT__</FormGroup>)).toMatchSnapshot();
  });
});
