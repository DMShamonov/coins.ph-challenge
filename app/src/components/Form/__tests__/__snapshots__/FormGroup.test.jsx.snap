// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`FormGroup\` component should render correctly with \`className\` prop 1`] = `
<div
  className="form__group __CLASSNAME__"
>
  __CONTENT__
</div>
`;

exports[`\`FormGroup\` component should render correctly with default props 1`] = `
<div
  className="form__group"
>
  __CONTENT__
</div>
`;
